SDK/AppCenterPush/Microsoft.AppCenter.Push.UWP/Push.cs

```diff
@@ -56,6 +61,20 @@ public void InstanceCheckLaunchedFromNotification(string args)
             }
         }
 
+        public void OnUserIdUpdated(object sender, UserIdUpdatedEventArgs e)
+        {
+            lock (UWPPushLock)
+            {
+                if (!string.IsNullOrEmpty(LatestPushToken))
+                {
+                    var pushInstallationLog = new PushInstallationLog(null, null, LatestPushToken, Guid.NewGuid(), e.userId);
```

Comment:
This does not lock on the same lock as the other function doing that. Plus it does not update LatestPushToken.

Reply:
`using (_mutex.GetLock(state))` is used elsewhere.

Reply:
@guperrot For 'This does not lock on the same lock as the other function doing that.', do you mean I should not lock here since the other function is locking it?
I don't quite understand what you mean by 'Plus it does not update LatestPushToken.', can you explain more?

Reply:
@achocron Do you mean since this function is invoked in the function that calls `using (_mutex.GetLock(state))`, I should not lock in this function?

Reply:
we should always use mutex when working at instance level on a field that can be accessed from multiple threads